woocommerce/woocommerce
# Add Delete variation item to the editor actions menu

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40598



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes
4. Wait until variations get generated
5. Then publish the product and save its `productId` and one `variationId` you like
6. Then go to `/wp-admin/admin.php?page=wc-admin&path=/product/{productId}/variation/{variationId}&tab=general` (replace the `{productId}` and `{variationId}` with the values picked in point 5)
7. On the top right corner of the screen the more menu should be shown with a new item called `Delete variation` 
<img width="794" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/615c2bb3-cb22-4418-8efa-69b55a8c9f33">

8. When click it, e confirmation modal should be shown 
<img width="715" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/e233d994-9bc8-4796-952b-ed3629368767">

9. I you click delete button from the modal then right after the variation is deleted a notice should be shown at the bottom left corner of the screen saying the that the variation was deleted and you should be redirected to the variation parent product page.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
